core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java

```diff
@@ -288,6 +302,45 @@ public ImageCatalog update(ImageCatalog source) {
         return create(imageCatalog);
     }
 
+    public Images getApplicableImages(String stackName) throws CloudbreakImageCatalogException {
+        return getApplicableImages(CLOUDBREAK_DEFAULT_CATALOG_NAME, stackName);
+    }
+
+    public Images getApplicableImages(String imageCatalogName, String stackName) throws CloudbreakImageCatalogException {
+        IdentityUser user = authenticatedUserService.getCbUser();
+        Stack stack = stackService.getByName(stackName, user);
+
+        if (!Status.AVAILABLE.equals(stack.getStackStatus().getStatus()) || !Status.AVAILABLE.equals(stack.getCluster().getStatus())) {
+            throw new BadRequestException("To retrieve list of images for upgrade both stack and cluster have to be in AVAILABLE state");
+        }
+
+        String currentImageUuid = getCurrentImageUuid(stack);
+        StatedImages statedImages = getImages(imageCatalogName, stack.cloudPlatform());
+        List<Image> filteredBaseImages =
+                filterByApplicability(imageCatalogName, statedImages.getImageCatalogUrl(), stack, statedImages.getImages().getBaseImages(), currentImageUuid);
+        List<Image> filteredHdpImages =
+                filterByApplicability(imageCatalogName, statedImages.getImageCatalogUrl(), stack, statedImages.getImages().getHdpImages(), currentImageUuid);
+        List<Image> filteredHdfImages =
+                filterByApplicability(imageCatalogName, statedImages.getImageCatalogUrl(), stack, statedImages.getImages().getHdfImages(), currentImageUuid);
+        return new Images(filteredBaseImages, filteredHdpImages, filteredHdfImages, statedImages.getImages().getSuppertedVersions());
```

Comment:
I think we should figure out here if it's a HDP or HDF cluster and don't iterate through both kind of prewarmed images if not necessary.